src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceClient.cs

```diff
@@ -76,7 +81,8 @@ public ResourcesClient(AzureContext context)
             : this(
                 AzureSession.ClientFactory.CreateClient<ResourceManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager),
                 new GalleryTemplatesClient(context),
-                AzureSession.ClientFactory.CreateClient<AuthorizationManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager))
+                AzureSession.ClientFactory.CreateClient<AuthorizationManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager),
```

Comment:
I noticed the on-demand failures because of this (bundling subscriptionClient with ResourcesClient). Let's not do this anymore. The bundling of other clients with ResourceManagementClient (like the gallery client) has been known to cause issues for us in the past (for instance, when running our cmdlets in a non prod environment, where gallery service may not necessarily be available).

Let's create a separate SubscriptionClient class, instantiate it like below (ClientFactory.CreateClient<>()) and use it in the location cmdlet. This will also ensure that you won't have to fix the failing tests in the on-demand job (and you shouldn't, since there is no need to create one for those who don't call the subscriptions api).